src/transformers/models/bloom/causal_message_passing.py

```diff
@@ -19,9 +27,24 @@ def build_message_passing_matrices(
         language model blocks of an autoregressive language model
     """
     message_passing_dicts = []
-    for t_ids, edge_sequence in zip(token_ids, edge_sequences):
-        message_passing_dict = {'tokens2edges': [], 'edges2tokens': [], 'inverse_edge_index': []}
+    for edge_sequence in edge_sequences:
+        message_passing_dict = {'tokens2edges': [], 'edges2tokens': [], 'edge_index': []}
         node2edge_idxs = defaultdict(list)
+        pred_node, edge, succ_node = edge_sequence[-1]
+        if isinstance(succ_node, SequenceElement):
+            sequence_end_idx = succ_node.end_idx
+        elif isinstance(edge, SequenceElement):
+            sequence_end_idx = edge.end_idx
+        else:
+            sequence_end_idx = pred_node.end_idx
+        pred_node, edge, succ_node = edge_sequence[0]
+        if isinstance(succ_node, SequenceElement):
+            sequence_start_idx = succ_node.end_idx
+        elif isinstance(edge, SequenceElement):
+            sequence_start_idx = edge.end_idx
+        else:
+            sequence_start_idx = pred_node.end_idx
```

Comment:
```suggestion
        if succ_node is not None:
            sequence_end_idx = succ_node.end_idx
        elif edge is not None:
            sequence_end_idx = edge.end_idx
        else:
            sequence_end_idx = pred_node.end_idx
        pred_node, edge, succ_node = edge_sequence[0]
        if succ_node is not None:
            sequence_start_idx = succ_node.end_idx
        elif edge is not None:
            sequence_start_idx = edge.end_idx
        else:
            sequence_start_idx = pred_node.end_idx
```
If this is correct, I find it more intuitive. Otherwise I had to go look into the code to answer the question of why are these checks mutually exclusive.